## Overview
This script contains a number of heuristics that, when applied to copy number data from different pipelines, aims to determine whether the sample has had a whole genome duplication or not. 

 * Large homozygous deletions
 * A cluster at 1/2 * CCF of the clonal cluster
 * Fraction of the genome in odd numbered copy number states when fit as tetraploid. These are also known as 50/50 segments after them having a CCF of 0.5 when the profile is fit as diploid
 
Based on these metrics and careful manual curation a couple of cutoffs have been defined. These metrics are applied to each profile individually, resulting in a call:

 * no\_correction
 * correct\_to\_wgd
 * correct\_to\_diploid
 * NA
 
## Known issues
 * The purities input table does not contain DKFZ output. The script therefore expects a separate file with the adjustments made by Jeffs method.
 * When no existing metrics are triggered and multiple methods call the profile diploid and multiple call it tetraploid there is no solution. These samples will be marked with NAs and need further investigation.
 * Metric 4 (50/50 segments) seems to be triggered quite often. Perhaps the threshold needs adjustment.

## Output format
The output of this script first iterates the input correction that it was given. Then it lists for every method what correction should be applied (as listed above), it lists which metrics have been triggered for that sample, the median and uncorrected purities from the input and a consensus purity call.

For now there is no consensus purity call as this bit of the code requires the purity from the DKFZ pipeline, which is not available in the corrent format. The code has been implemented for this feature, but commented out and resides around line 200 of the icgc\_resolve\_wgd.R script.

| samplename   |   input\_broad   |  input\_dkfz   |   input\_mustonen095    |   input\_peifer  |  input\_vanloo\_wedge   |   corrections\_broad    |   corrections\_dkfz  |      corrections\_mustonen095 corrections\_peifer   |   corrections\_vanloo\_wedge    |    metric\_2    |    metric\_3   |     metric\_4   |     metric\_5    |    metric\_6.1   |   metric\_6.2  |    purity\_median |  uncorrected\_purities  |  consensus\_purity |
|-----|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|:----:|
| 003819bc-c415-4e76-887c-931d60ed39e7 |   None  |  None  |  None  |  None  |  None  |  NA    |  NA   |   NA   |   NA   |   NA   |   FALSE  | FALSE  | FALSE |  FALSE |  FALSE  | FALSE |  0.727802739726 | 0.609391638466,0.7104,0.774146,0.745205479452 | NA |


## Required input data
This script relies on a couple of different input files of which the format is supplied below.

### Purities and adjustments
Overview file with adjustments and purities generated by Jeff. Note that the example below does not contain the DKFZ output as at the time that was not available. The script therefore needs another file that includes the DKFZ corrections, as is outlined below.

Expected format:

| tumor  |  purity\_consensus  |  purity\_battenberg | purity\_median | purity\_mean | methods | corrections | uncorrected\_purities | corrected\_purities | uncorrected\_purities\_stdev |   corrected\_purities\_stdev |  genome\_consensus\_proportion |
|--------|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|
| 0009b464-b376-4fbc-8a56-da538269a02f |   0.92456  |  0.92456  |  0.88865  |  0.89374  |  broad,mustonen095,peifer,vanloo_wedge  |  None,None,None,None | 0.87312493848,0.8873,0.89,0.92456  |  0.87312493848,0.8873,0.89,0.92456  |  0.018909747526  |  0.018909747526  |  0.131110216084 |
| 003819bc-c415-4e76-887c-931d60ed39e7  |  0.745205479452 |   0.745205479452  |  0.727802739726 |   0.70978577948   | broad,mustonen095,peifer,vanloo_wedge | None,None,None,None |   0.609391638466,0.7104,0.774146,0.745205479452  |  0.609391638466,0.7104,0.774146,0.745205479452  |  0.0622015955348  |  0.0622015955348 |  0.828143348583 |

### Adjustments for all methods
As the original purities and adjustments files did not contain the adjustments for the DKFZ pipeline a separate file was later created that contains all the adjustment information. At the moment the pipeline needs this extra file.

Expected format (no header):

| | |
|-----------|-----------|
| 003819bc-c415-4e76-887c-931d60ed39e7  |  broad=None,dkfz=None,mustonen095=None,peifer=None,vanloo_wedge=None |
| 00493087-9d9d-40ca-86d5-936f1b951c93  |  broad=None,dkfz=None,mustonen095=None,peifer=None,vanloo_wedge=None |
| 00508f2b-36bf-44fc-b66b-97e1f3e40bfa  |  broad=None,dkfz=None,mustonen095=None,peifer=None,vanloo_wedge=None |
| 005794f1-5a87-45b5-9811-83ddf6924568  |  broad=None,dkfz=None,mustonen095=None,peifer=None |

### Copy number
Copy number data as output by Jeffs clonal consensus method. There are two variants (in the same format): Diploid consensus with adjustment where tetraploid profiles are adjusted downwards and tetraploid consensus where only the tetraploid samples are used to create consensus without adjustment.

Expected format (PCAWG-11 calibration format):

| chromosome  |    start  | end   |  copy\_number   |  major\_cn    |    minor\_cn    |    clonal\_frequency |
|-----------|:-----------:|:-----------:|:-----------:|:-----------:|:-----------:|:-----------:|
| 1   |    123 | 456 | 2  |     1   |    1    |   1.0 |
| 1   |    457 | 789 |  2  |     1    |   1    |   1.0 |
| 1   |    1000   |     1200    |    2  |     1   |    1    |   1.0 |

### Clustering results
Run clustering on the diploid consensus where tetraploid calls have been adjusted downwards. If this adjustment is incorrect then in some cases this will give rise to a mutation cluster at 1/2 * CCF of the clonal cluster. These data support a metric that incorporates that.

Expected format (PCAWG-11 calibration format):

| cluster | n_ssms | proportion |
|---------|:------:|:----------:|
| 1 | 141.0 | 0.186 |
| 2 | 59.0 | 0.748 |