


## Required input data
This script relies on a couple of different input files of which the format is supplied below.

### Adjustments and purities
Overview file with adjustments and purities generated by Jeff

| tumor  |  purity\_consensus  |  purity\_battenberg | purity\_median | purity\_mean | methods | corrections | uncorrected\_purities | corrected\_purities | uncorrected\_purities\_stdev |   corrected\_purities\_stdev |  genome\_consensus\_proportion |
|--------|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|:--------:|
| 0009b464-b376-4fbc-8a56-da538269a02f |   0.92456  |  0.92456  |  0.88865  |  0.89374  |  broad,mustonen095,peifer,vanloo_wedge  |  None,None,None,None | 0.87312493848,0.8873,0.89,0.92456  |  0.87312493848,0.8873,0.89,0.92456  |  0.018909747526  |  0.018909747526  |  0.131110216084 |
| 003819bc-c415-4e76-887c-931d60ed39e7  |  0.745205479452 |   0.745205479452  |  0.727802739726 |   0.70978577948   | broad,mustonen095,peifer,vanloo_wedge | None,None,None,None |   0.609391638466,0.7104,0.774146,0.745205479452  |  0.609391638466,0.7104,0.774146,0.745205479452  |  0.0622015955348  |  0.0622015955348 |  0.828143348583 |

### Copy number
Copy number data as output by Jeffs clonal consensus method. There are two variants (in the same format): Diploid consensus with adjustment where tetraploid profiles are adjusted downwards and tetraploid consensus where only the tetraploid samples are used to create consensus without adjustment.

| chromosome  |    start  | end   |  copy\_number   |  major\_cn    |    minor\_cn    |    clonal\_frequency |
|-----------|:-----------:|:-----------:|:-----------:|:-----------:|:-----------:|:-----------:|
| 1   |    123 | 456 | 2  |     1   |    1    |   1.0 |
| 1   |    457 | 789 |  2  |     1    |   1    |   1.0 |
| 1   |    1000   |     1200    |    2  |     1   |    1    |   1.0 |

### Clustering results
Run clustering on the diploid consensus where tetraploid calls have been adjusted downwards. If this adjustment is incorrect then in some cases this will give rise to a mutation cluster at 1/2 * CCF of the clonal cluster. These data support a metric that incorporates that.

Expected format:
| cluster | n_ssms | proportion |
|---------|:------:|:----------:|
| 1 | 141.0 | 0.186 |
| 2 | 59.0 | 0.748 |